### ISSUE TITLE
Required rule with scalar fields / 0 value

### ISSUE THREAD

**Issue by lsgndln** [bufbuild/protovalidate-go#270]
Hello, I wanted to ask a question about how are handled the scalar fields validation.

I use `syntax="proto3"`, and have an int32 field, on which I added `required`.

```
message MyMessage {
  int32 my_field = 1 [(buf.validate.field).required = true];
}

From the docs, I saw this :

```
Fields that don't track presence are always validated by Protovalidate, whether they are set or not. It is not necessary to add required:
```

From this and doc I read, I understand, that since I use the proto3 syntax, this field will always validated, because it will always have a value. So `required` does not make sense to add in a way.

However, if I set my field `my_field` to 0 (its default value), the validator is complaining that `my_field` is required ?

Again, I understand that adding `required` to it is not necessary, but why the default value is not accepted? 

Thank you
